server/sonar-server/src/main/java/org/sonar/server/organization/ws/CreateAction.java

```diff
@@ -110,118 +92,45 @@ public void handle(Request request, Response response) throws Exception {
     String name = wsSupport.getAndCheckMandatoryName(request);
     String requestKey = getAndCheckKey(request);
     String key = useOrGenerateKey(requestKey, name);
-    wsSupport.getAndCheckDescription(request);
-    wsSupport.getAndCheckUrl(request);
-    wsSupport.getAndCheckAvatar(request);
+    String description = wsSupport.getAndCheckDescription(request);
+    String url = wsSupport.getAndCheckUrl(request);
+    String avatar = wsSupport.getAndCheckAvatar(request);
 
     try (DbSession dbSession = dbClient.openSession(false)) {
-      checkKeyIsNotUsed(dbSession, key, requestKey, name);
-
-      OrganizationDto organization = createOrganizationDto(dbSession, request, name, key);
-      GroupDto group = createOwnersGroup(dbSession, organization);
-      createDefaultTemplate(dbSession, organization, group);
-      addCurrentUserToGroup(dbSession, group);
-
-      dbSession.commit();
+      OrganizationDto organization = organizationCreation.create(
+        dbSession,
+        userSession.getUserId().longValue(),
+        newOrganizationBuilder()
+          .setName(name)
+          .setKey(key)
+          .setDescription(description)
+          .setUrl(url)
+          .setAvatarUrl(avatar)
+          .build());
 
       writeResponse(request, response, organization);
+    } catch (OrganizationCreation.KeyConflictException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)